File: plugins/hello-python/validate.py
import json
import sys


def main() -> None:
    payload = json.loads(sys.stdin.read() or "{}")
    source_name = str(payload.get("workspaceConfig", {}).get("sourceName", "")).strip()
    errors = []

    if not source_name:
        errors.append({"field": "sourceName", "message": "sourceName is required"})

    sys.stdout.write(json.dumps({"valid": len(errors) == 0, "errors": errors}))


if __name__ == "__main__":
    main()

File: plugins/hello-node/validate.mjs
import process from "node:process";

async function readJSON() {
  const chunks = [];
  for await (const chunk of process.stdin) {
    chunks.push(chunk);
  }

  const raw = Buffer.concat(chunks).toString("utf8").trim();
  return raw ? JSON.parse(raw) : {};
}

const payload = await readJSON();
const errors = [];

if (!String(payload.workspaceConfig?.sourceName ?? "").trim()) {
  errors.push({
    field: "sourceName",
    message: "sourceName is required",
  });
}

process.stdout.write(
  JSON.stringify({
    valid: errors.length === 0,
    errors,
  }),
);
